### DISCUSSION TITLE
Can't figure out how to use setPersistence

### DISCUSSION THREAD

**Discussion by grenierdev** [firebase/firebase-js-sdk#4670] [Q&A] [answered]
I can't figure out to properly import persistence other then `inMemoryPersistence`. 
```ts
import type { User, Auth } from 'firebase/auth';
import { signInWithEmailAndPassword, onAuthStateChanged, setPersistence, signOut, inMemoryPersistence, browserLocalPersistence } from 'firebase/auth';
```
I'm using Snowpack and I'm getting this error
```
The requested module '../../_snowpack/pkg/firebase.auth.v0.900.19.js' does not provide an export named 'browserLocalPersistence'
```
Obviously, I'm doing something wrong. How do we import the persistence `'LOCAL'` or `'SESSION'` ?

**Comment by akauppi** [comment]
Have you considered that there would be no importing. Based on [this](https://modularfirebase.web.app/reference/auth.persistence.type) they are strings.

What’s confusing is the sample showing `browserLocalPersistence`. It should be just `”LOCAL”`.

Disclaimer: did not try.

**Reply by grenierdev** [reply]
The `Persistence` is an interface. Both `inMemoryPersistence` and `browserLocalPersistence` implements `Persistence`. There's actually a few to choose from according to the typings : `inMemoryPersistence` (for `'NONE'`), `browserLocalPersistence` (`LOCAL`), `browserSessionPersistence` (`SESSION`) and `indexedDBLocalPersistence` (`LOCAL`).

But they are not exported from the auth module.

**Comment by grenierdev** [comment, marked as answer]
I've switch to Vitejs and everything is working fine. I suspect that Snowpack doesn't use the `"browser"` main file of the package.json.

**Comment by Nikhil-gna** [comment]
Firebase Modular Version V9
```
import {
  getAuth,
  setPersistence,
  browserLocalPersistence,
  browserSessionPersistence,
  inMemoryPersistence
} from "firebase/auth";

const auth = getAuth()

await setPersistence(auth, browserLocalPersistence);
```

Namespaced Version V8 -  ```firebase.auth.Auth.Persistence.LOCAL  ```  
    
   Modular Version V9 -    ```browserLocalPersistence```
